Ex14

ruminmaxim opened the pull request:
1. Calculator with specified functions.
2. Game Hangman

ruminmaxim committed:
1. Calculator with specified functions.
2. Game Hangman

eugrdn commented on src/ex14_js-context/task-02.js:
```diff
@@ -0,0 +1,68 @@
+function Hangman(word) {
+    this.secretWord = word.toLowerCase();
```
2-5 испоьзуются в startAgain -- вынеси в отдельный метод

eugrdn commented on src/ex14_js-context/task-01.js:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    this.add = function (number) {
+        if (number !== undefined) {
```
для калькулятора не совсем корректно, посмотри https://learn.javascript.ru/number#proverka-na-chislo

eugrdn commented on src/ex14_js-context/task-02.js:
```diff
@@ -0,0 +1,68 @@
+        check = letter.toLowerCase();
+        if (this.secretWord.indexOf(check) != -1) {
+            var userWordCreating = [];
+            for (i = 0; i < this.secretWord.length; i++) {
```
попробуй здесь использовать `.reduce`

ruminmaxim committed:
working

ruminmaxim committed:
1. Check for number was added to the calculator
2.1. The code was shorted by adding extra method
2.2. Checking of letter was made by Reduce Method

ruminmaxim committed:
remove html

ruminmaxim committed:
Check made by reduce method

eugrdn commented on src/ex14_js-context/task-01.js:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    this.multiply = function (number) {
+        if (isNumeric(number) === true) {
```
здесь неявно происходит приведения того, что вернет `isNumeric(number)` к Boolean, `=== true` проверка излишняя

ruminmaxim committed:
Code is fixed ("===true" was deleted)
